diff -ru dieharder.orig/Makefile dieharder/Makefile
--- dieharder.orig/Makefile	2006-07-27 11:43:30.000000000 -0500
+++ dieharder/Makefile	2006-09-11 20:23:29.000000000 -0500
@@ -45,6 +45,7 @@
 SOURCE = $(PROGRAM:=.c)
 SOURCES = $(SOURCE) $(TSOURCES) \
     add_my_types.c \
+    gnu_r_rngs.c \
     dev_random.c \
     dev_urandom.c \
     empty_random.c \
diff -ru dieharder.orig/add_my_types.c dieharder/add_my_types.c
--- dieharder.orig/add_my_types.c	2006-07-12 11:41:25.000000000 -0500
+++ dieharder/add_my_types.c	2006-09-11 20:22:04.000000000 -0500
@@ -30,6 +30,12 @@
 GSL_VAR const gsl_rng_type *gsl_rng_empty_random;
 GSL_VAR const gsl_rng_type *gsl_rng_file_input;
 GSL_VAR const gsl_rng_type *gsl_rng_file_input_raw;
+GSL_VAR const gsl_rng_type *gsl_rng_r_wichmann_hill;	/* edd Aug 2006 */
+GSL_VAR const gsl_rng_type *gsl_rng_r_marsaglia_mc;	/* edd Aug 2006 */
+GSL_VAR const gsl_rng_type *gsl_rng_r_super_duper;	/* edd Aug 2006 */
+GSL_VAR const gsl_rng_type *gsl_rng_r_mersenne_twister;	/* edd Sep 2006 */
+GSL_VAR const gsl_rng_type *gsl_rng_r_knuth_taocp;	/* edd Sep 2006 */
+GSL_VAR const gsl_rng_type *gsl_rng_r_knuth_taocp2;	/* edd Sep 2006 */
 
 void add_my_types()
 {
@@ -60,5 +66,23 @@
  types[i] = (gsl_rng_file_input_raw);
  if(verbose) printf("# add_my_types():  Added type %d = %s\n",i,types[i]->name);
  i++;
+ types[i] = (gsl_rng_r_wichmann_hill);
+ if(verbose) printf("# add_my_types():  Added type %d = %s\n",i,types[i]->name);
+ i++;
+ types[i] = (gsl_rng_r_marsaglia_mc);
+ if(verbose) printf("# add_my_types():  Added type %d = %s\n",i,types[i]->name);
+ i++;
+ types[i] = (gsl_rng_r_super_duper);
+ if(verbose) printf("# add_my_types():  Added type %d = %s\n",i,types[i]->name);
+ i++;
+ types[i] = (gsl_rng_r_mersenne_twister);
+ if(verbose) printf("# add_my_types():  Added type %d = %s\n",i,types[i]->name);
+ i++;
+ types[i] = (gsl_rng_r_knuth_taocp);
+ if(verbose) printf("# add_my_types():  Added type %d = %s\n",i,types[i]->name);
+ i++;
+ types[i] = (gsl_rng_r_knuth_taocp2);
+ if(verbose) printf("# add_my_types():  Added type %d = %s\n",i,types[i]->name);
+ i++;
  
 }
Only in dieharder: gnu_r_rngs.c
